problem_builder/public/js/mentoring_with_steps.js

```diff
@@ -0,0 +1,157 @@
+function MentoringWithStepsBlock(runtime, element) {
+
+    var steps = runtime.children(element).filter(
+        function(c) { return c.element.className.indexOf('pb-mentoring-step') > -1; }
+    );
+    var step = $('.mentoring', element).data('step');
+    var active_child, checkmark, submitDOM, nextDOM, tryAgainDOM, submitXHR;
+
+    function isLastChild() {
+        return (active_child === steps.length-1);
+    }
+
+    function updateStep() {
+        var handlerUrl = runtime.handlerUrl(element, 'update_step');
+        $.post(handlerUrl, JSON.stringify(step+1))
```

Comment:
@itsjeyd nit: it actually does slightly more than the name tells - it also advances to the next step. Consider replacing `step+1` with parameter.